### ISSUE TITLE
The scoring / follower counter system

### ISSUE THREAD

**Issue by FireWaterGasoline** [kirijiri/GalactosIntolerant#37]
This is one of them things we need for the Insomnia build. 

The population death is going to be dynamic: as soon as you disrupt a planet's path, people start  dying at a certain rate. The more you disrupt it from the starting conditions (speed / direction) the higher the mortality rate becomes. This doesn't cancel the fact that population also dies in bursts during holding, flicking and gravity beam activation. 

Example: The lava planet has a starting population of 5 000 000. The player stops it for 5 seconds and 50 000 people perish. But the planet is also now behind on it's seasonal cycle and gains a 100 people per second death rate. Player flicks it in opposite direction of it's original course (100 000 people die) - the planet acquires additional 400 people per second death rate. So death rate is now at 500 per second. This only get's worse. Obviously this is dynamically reflected in the follower count which is a % of the total population of the system. 

The above numbers are for demonstration purposes only. The death rate modifiers should ideally be in tinker. 

![ferrrrret](https://cloud.githubusercontent.com/assets/8594551/6770447/f2129c4a-d0b7-11e4-80e3-83cf2308c7ce.gif)

**Comment by kirijiri** [comment]
what's in the game right now is a death percentage...
since you originally said that hold doesn't kill anyone, it gets triggered
only by flick... you flick a planet % people die (that percentage is from
the Max population, not the current one)... its in tinker. that rate
doesn't change.
so you want an individual rate for each planet? they will die differently
fast?

**Comment by FireWaterGasoline** [comment]
Yes and yes. 

The important bit is that if only one planet is interacted with, only one planet gets the per second death rate. If two - just those two, etc. The planets will die differently fast not necessarily because they have different default modifiers for their rates but because they have been interacted with differently. 

Also, I don't think I've mentioned it but if the direction of movement is reversed, just double the death rate because that's even worse than just moving slower or faster - now your seasons are reversed! 

Ideally, if the player manges to get the planet back roughly to its original velocity, the death rate should also go down. Maybe the most helpful way of thinking about this would be to compare this mechanic to that game where one person hides an object and the other one is looking for it, the closer they are to it, the one who hid it goes "cold, cold, warm, warmer, warmer, hot, super hot, oh god I'm buring! ahaghghgh!!!" and then back to "cooler, cold, freezing" if they start moving the wrong way : ) 


Btw, holding should also kill. There must have been some miscommunication. I apologise if I caused this. Here's another ferret gif as a gesture of good will

![ferret in a box](https://cloud.githubusercontent.com/assets/8594551/6778568/957ca90a-d14b-11e4-8916-4cab5fcf5b34.gif)

**Comment by kirijiri** [comment]
Ok, so just to confirm I got this correct..

planet gets initial population (lets say 5 000 000)
planet gets death percentage for action (of initial population, lets say
10% = 500000)
planet gets death percentage for bleed (of initial population, lets say 5%
= 250000)

planet gets held x seconds: kill how many people per second?
planet gets flicked x/y distance/time: kill how many people per intensity?

then there is an additional bleed for how long? or Ed said that the bleed
disappears when it matches the initial speed and direction again, how much
is there supposed to be when its 10% faster than the inital speed or when
its in the opposite direction?

Also... as a scientific note... seasons are not reversed if the planet
rotates the other way around... leaves will not go from dead to dying to
green to blossoming.. thats now how it works :p

Also, I take aaaaaaaall the ferret gifs :p... they are awesome

**Comment by FireWaterGasoline** [comment]
Yep, you got it right. 

Can you just put whatever numbers and tie them to tinker? Follow your heart for the initial ones : )

I don't understand "additional bleed". What's that? 


Re scientific note: though it would be cool if leaves reversed their life cycle, that's not what I was talking about. Take Earth for example. It's the beginning of spring now in the northern hemisphere. If the direction of Earth's rotation around the sun were to be reversed right now (and if we ignore all other side effects) it would start getting colder and colder every day. We would go back to winter instead of more spring and then summer. So you could expect to get roughly three more months of winter after the first month of spring. And then it would be spring again, eventually, of course, but only after another winter. 

![ferret warming up](https://cloud.githubusercontent.com/assets/8594551/6780241/56b84a34-d15b-11e4-97a3-b7442d002df3.gif)

**Comment by disusedgenius** [comment]
To be fair, it would get colder and colder because time would be reversed
and we'd be going back into February.

**Comment by FireWaterGasoline** [comment]
![nailed it](https://cloud.githubusercontent.com/assets/8594551/6780367/78fa8ed0-d15c-11e4-984f-077fc97e855e.png)

**Comment by kirijiri** [comment]
right now it'd go back to being winter, yes, but  then autumn would turn
into spring and summer would turn into summer again... if the direction
change would happen in summer, itd still go over to autumn (which used to
be spring) and then winter... the transitions would still be the same,
since nature wouldn't work any reversed way....

and the additional bleed... you said " gains a 100 people per second death
rate" and "Ideally, if the player manges to get the planet back roughly to
its original velocity, the death rate should also go down. " .... which
sounds like you want an initial kill rate and another constant one over
time until it possibly gets back to its natural speed/direction....
so its 2 kinds of damage.. one for hold/flick and one forever.. right?

**Comment by FireWaterGasoline** [comment]
Yes, if you were in the middle of summer or winter, it would be the same thing even if you were to reverse a direction. I agree : ) 

And yes, that is the rightest ever! 

![ferret cleaner](https://cloud.githubusercontent.com/assets/8594551/6785158/ed4efdf6-d17b-11e4-9413-0b333857a3c3.gif)
